Source/Portal/Web/layouts/ActionBarState.ts

```diff
@@ -21,15 +21,27 @@ export class ActionBarState {
         _messenger
             .observe(ActionBarStructureChanged)
             .subscribe((_) => {
-                this.current.next(_.actionBar as ActionBar | undefined);
+
+                if(_.actionBar.button != ActionBarActionForMessage.EMPTY) {
+                    const actionBar = {
+                        placement: _.actionBar.placement,
+                        button: new ActionBarAction(
+                            _.actionBar.button.text,
+                            _.actionBar.button.icon,
+                            this._publishActionTriggered.bind(this),
+                            _.actionBar.button.id,
+                            )
+                    } as ActionBar;
+                    this.current.next(actionBar);
+                }
             });
```

Comment:
You could use the RXJS operator `filter`to avoid the if statement. Then your subscribe will only get the objects it is expecting:

```typescript
        _messenger
            .observe(ActionBarStructureChanged)
            .pipe(filter(_ => _.actionBar.button !== ActionBarActionForMessage.EMPTY))
            .subscribe((_) => {
                const actionBar = {
                    placement: _.actionBar.placement,
                    button: new ActionBarAction(
                        _.actionBar.button.text,
                        _.actionBar.button.icon,
                        this._publishActionTriggered.bind(this),
                        _.actionBar.button.id,
                    )
                } as ActionBar;
                this.current.next(actionBar);
            });
```

Reply:
makes sense! Nice to use `filter` as well